src/main/java/org/jitsi/utils/dsi/DominantSpeakerIdentification.java

```diff
@@ -431,6 +482,96 @@ private synchronized Speaker getOrCreateSpeaker(T id)
         return speaker;
     }
 
+    private synchronized void updateLoudestList(Speaker speaker, int level, long now)
+    {
+        long oldestValid = now - energyExpireTimeMs;
+
+        /* exponential smoothing */
+        speaker.energyScore = (energyAlphaPct * level + (100 - energyAlphaPct) * speaker.energyScore) / 100;
+
+        logger.debug("Want to add " + speaker.id.toString() + " with score " + speaker.energyScore + ". Last level = " + level + ".");
+
+        int i = 0;
+        while (i < loudest.size())
+        {
+            Speaker cur = loudest.get(i);
+            if (cur.getLastLevelChangedTime() < oldestValid)
+            {
+                logger.debug("Removing " + cur.id.toString() + ". old.");
+                loudest.remove(i);
+                continue;
+            }
+            if (cur == speaker)
+            {
+                logger.debug("Removing " + cur.id.toString() + ". same.");
+                loudest.remove(i);
+                continue;
+            }
+            ++i;
+        }
+
+        i = 0;
+        while (i < loudest.size())
+        {
+            Speaker cur = loudest.get(i);
+            if (cur.energyScore < speaker.energyScore)
+                break;
+            ++i;
+        }
+
+        if (i < numLoudestToRoute)
+        {
+            logger.debug("Adding " + speaker.id.toString() + " at position " + i + ".");
+            loudest.add(i, speaker);
+        }
+
+        while (loudest.size() > numLoudestToRoute)
+            loudest.remove(numLoudestToRoute);
+
+        i = 0;
+        while (i < loudest.size())
```

Comment:
Surround this look in `if (logger.isDebugEnabled())`?

Reply:
done, though these logs are probably temporary.